hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheImpl.java

```diff
@@ -84,12 +85,13 @@ public NearCacheImpl(String mapName, NodeEngine nodeEngine, SizeEstimator<NearCa
 
         Config config = nodeEngine.getConfig();
         NearCacheConfig nearCacheConfig = config.findMapConfig(mapName).getNearCacheConfig();
+        EvictionConfig evictionConfig = nearCacheConfig.getEvictionConfig();
         this.inMemoryFormat = nearCacheConfig.getInMemoryFormat();
-        this.maxSize = nearCacheConfig.getMaxSize() <= 0 ? Integer.MAX_VALUE : nearCacheConfig.getMaxSize();
+        this.maxSize = evictionConfig.isSizeConfigured() ? evictionConfig.getSize() : NearCacheConfig.DEFAULT_MAX_SIZE;
```

Comment:
This is the fallback to the old `DEFAULT_MAX_SIZE`, if the size was not configured explicitly.